### ISSUE TITLE
pylance detect error that should not detect in matplotlib (3d)

### ISSUE THREAD

**Issue by victorballester7** [microsoft/pyright#6690]
Hi everyone! I'm in VSCode and I have the extension from Microsoft for Python. I have the following error. Consider the following code without errors (a priori).

```
import matplotlib.pyplot as plt
import numpy as np

x = [74, 74, 74, 74, 74, 74, 74, 74, 192, 74]
y = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
z = np.zeros(10)
dx = np.ones(10)*10
dy = np.ones(10)
dz = [1455, 1219, 1240, 1338, 1276, 1298, 1292, 1157, 486, 1388]

fig = plt.figure()
ax = fig.add_subplot(projection='3d')

bar3d = ax.bar3d(x, y, z, dx, dy, dz, color='C0')

ax.set_title("Data Analysis ")
ax.set_xlabel('x')
ax.set_ylabel('y')
ax.set_zlabel('z')

plt.show()
```
 
I am getting the following error from Pylance:

`Cannot access member "bar3d" for type "Axes"
  Member "bar3d" is unknown`

`Cannot access member "set_zlabel" for type "Axes"
  Member "set_zlabel" is unknown`

He does not detect that the figure is 3d. It's a trifle because it compiles well, but on the other hand it's very annoying if you want not to have any warning. Also, it does not detect the 3d properties in the intellisense, of course...

Does anyone know how to solve this? Thanks!

**Comment by rchiodo** [comment]
This is because of the stubs that matplotlib [has](https://github.com/matplotlib/matplotlib/blob/a2a28093ba836366e7ff0c6f1edb4fd98b47a04d/lib/matplotlib/figure.pyi#L83). The stub for `add_subplot` would likely need an overload for the literal of `projection='3d'`. You should log an issue on matplotlib.

Yeah, I can verify I can get the correct types for your example if I modify the stub like so:

```python
    @overload
    def add_subplot(self, projection: Literal['3d'], **kwargs) -> Axes3D: ...
```

I don't know if that's the correct modification or not though. The matplotlib team would know better.

**Comment by bschnurr** [comment]
matplotlib here https://github.com/matplotlib/matplotlib/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc

**Comment by victorballester7** [comment]
Thank you guys! I'll definitely will do it @rchiodo. By the way, your workaround @debonte works fine (if you add `from typing import cast`)!
